create_test.go

```diff
@@ -1059,3 +1059,50 @@ func TestCreateCreateContainer(t *testing.T) {
 		assert.NoError(err)
 	}
 }
+
+func TestCopyParentCPUSetFail(t *testing.T) {
+	assert := assert.New(t)
+
+	cgroupsPath, err := ioutil.TempDir(testDir, "cgroups-path-")
+	if err != nil {
+		t.Fatalf("Could not create temporary cgroups directory: %s", err)
+	}
+	defer os.RemoveAll(cgroupsPath)
+
+	err = copyParentCPUSet(cgroupsPath, testDir)
+	assert.Error(err)
+}
+
+func TestCopyParentCPUSetSuccessful(t *testing.T) {
+	if os.Geteuid() != 0 {
+		// Create cgroup directory must be root.
+		t.Skip(testDisabledNeedRoot)
+	}
+
+	assert := assert.New(t)
+
+	cgroupsRootPath := findCgroupRoot()
+	if cgroupsRootPath == "" {
+		t.Skip(testDisabledNeedCgroup)
+	}
+
+	cgroupsCPUSetRoot := filepath.Join(cgroupsRootPath, "cpu")
```

Comment:
This should be:

```Go
cgroupsCPUSetRoot := filepath.Join(cgroupsRootPath, "cpuset")
```

for the test to pass (as sudo).

Reply:
thanks for testing my code.  the test is using for testing copy data from /sys/fs/cgroup/cpu/cpuset.cpus and /sys/fs/cgroup/cpu/cpuset.mems to container's cgroup path, I test this on my redhat machine many times,  it can pass always, and now I check this on my ubuntu machine , it shows that 
```
root@www:/home/ace/gosrc/src/github.com/clearcontainers/runtime# ls /sys/fs/cgroup/cpu
cgroup.clone_children  cgroup.sane_behavior  cpuacct.usage         cpu.cfs_period_us  cpu.shares  docker      notify_on_release  system.slice  user.slice
cgroup.procs           cpuacct.stat          cpuacct.usage_percpu  cpu.cfs_quota_us   cpu.stat    init.scope  release_agent      tasks

```
looks like cgroup system on ubuntu and redhat is different, ubuntu not have cpuset.cpus and couset.mem in cpu cgroup path as redhat does
I really sorry for not test this on ubuntu machine, I will update my test

do I need to reopen a PR, or still use #643   @sameo

Reply:
I add more check for the test
```
diff --git a/create_test.go b/create_test.go
index 6e29d79..6089047 100644
--- a/create_test.go
+++ b/create_test.go
@@ -1091,6 +1091,16 @@ func TestCopyParentCPUSetSuccessful(t *testing.T) {
                t.Skip(testDisabledNeedCgroup)
        }
 
+       cpusetCpusPath := filepath.Join(cgroupsCPUSetRoot, "cpuset.cpus")
+       if _, err := os.Stat(cpusetCpusPath); err != nil && os.IsNotExist(err) {
+               t.Skip(testDisabledNeedCgroup)
+       }
+
+       cpusetMemsPath := filepath.Join(cgroupsCPUSetRoot, "cpuset.mems")
+       if _, err := os.Stat(cpusetMemsPath); err != nil && os.IsNotExist(err) {
+               t.Skip(testDisabledNeedCgroup)
+       }
+

```

Reply:
@Ace-Tang On both Ubuntu and Fedora (26) the cpu cgroup does not have a cpuset. I believe any distro with a fairly recent kernel (4.x) will have that behaviour. You see an empty cpuset on the cpu cgroup on older kernels (RHEL, Centos).